modifytags/modifytags.go

```diff
@@ -0,0 +1,367 @@
+		return tags, nil
+	}
+
+	for key, val := range m.RemoveOptions {
+		for _, option := range val {
+			tags.DeleteOptions(key, option)
+		}
+	}
+
+	return tags, nil
+}
+
+func (m *Mutation) addTagOptions(tags *structtag.Tags) (*structtag.Tags, error) {
+	if len(m.AddOptions) == 0 {
+		return tags, nil
+	}
+
+	for key, val := range m.AddOptions {
+		tags.AddOptions(key, val...)
+	}
+
+	return tags, nil
+}
+
+func (m *Mutation) addTags(fieldName string, tags *structtag.Tags) (*structtag.Tags, error) {
+	if len(m.Add) == 0 {
+		return tags, nil
+	}
+
+	splitted := camelcase.Split(fieldName)
+	name := ""
+
+	unknown := false
+	switch m.Transform {
+	case "snakecase":
+		var lowerSplitted []string
+		for _, s := range splitted {
+			s = strings.Trim(s, "_")
+			if s == "" {
+				continue
+			}
+			lowerSplitted = append(lowerSplitted, strings.ToLower(s))
+		}
+
+		name = strings.Join(lowerSplitted, "_")
+	case "lispcase":
+		var lowerSplitted []string
+		for _, s := range splitted {
+			lowerSplitted = append(lowerSplitted, strings.ToLower(s))
+		}
+
+		name = strings.Join(lowerSplitted, "-")
+	case "camelcase":
+		var titled []string
+		for _, s := range splitted {
+			titled = append(titled, strings.Title(s))
+		}
+
+		titled[0] = strings.ToLower(titled[0])
+
+		name = strings.Join(titled, "")
+	case "pascalcase":
+		var titled []string
+		for _, s := range splitted {
+			titled = append(titled, strings.Title(s))
+		}
+
+		name = strings.Join(titled, "")
+	case "titlecase":
+		var titled []string
+		for _, s := range splitted {
+			titled = append(titled, strings.Title(s))
+		}
+
+		name = strings.Join(titled, " ")
+	case "keep":
+		name = fieldName
+	default:
+		unknown = true
+	}
+
+	if m.ValueFormat != "" {
+		prevName := name
+		name = strings.ReplaceAll(m.ValueFormat, "{field}", name)
+		if name == m.ValueFormat {
+			// support old style for backward compatibility
+			name = strings.ReplaceAll(m.ValueFormat, "$field", prevName)
+		}
+	}
+
+	for _, key := range m.Add {
+		splitted = strings.SplitN(key, ":", 2)
+		if len(splitted) >= 2 {
+			key = splitted[0]
+			name = strings.Join(splitted[1:], "")
+		} else if unknown {
+			// the user didn't pass any value but want to use an unknown
+			// transform. We don't return above in the default as the user
+			// might pass a value
+			return nil, fmt.Errorf("unknown transform option %q", m.Transform)
+		}
+
+		tag, err := tags.Get(key)
+		if err != nil {
+			// tag doesn't exist, create a new one
+			tag = &structtag.Tag{
+				Key:  key,
+				Name: name,
+			}
+		} else if m.Override {
+			tag.Name = name
+		}
+
+		if err := tags.Set(tag); err != nil {
+			return nil, err
+		}
+	}
+
+	return tags, nil
+}
+
+func isPublicName(name string) bool {
+	for _, c := range name {
+		return unicode.IsUpper(c)
+	}
+	return false
+}
+
+// Rewrite rewrites the node for structs between the start and end lines
+func (m *Mutation) Rewrite(fset *token.FileSet, node ast.Node, start, end int) (ast.Node, error) {
```

Comment:
This should either be inlined into Apply (see discussion below), or moved right below Apply, so that the logic can more easily be read from top-to-bottom.